pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -129,11 +129,16 @@ pub mod pallet {
 		/// The maximum number of currencies allowed for a single pool.
 		#[pallet::constant]
 		type MaxCurrencies: Get<u32>;
-		/// The deposit required for each bonded currency.
 
 		#[pallet::constant]
 		type MaxStringLength: Get<u32>;
 
+		/// The maximum denomination that bonded currencies can use. To ensure
+		/// safe operation, this should be set so that 10^MaxDenomination <
```

Comment:
If you want to make sure that only these values are specified here, you can add an integrity test (available with the `try-runtime` feature) that only checks the values of the `Config` implementations for any given runtime, according to the specified rules. You can check an example here: https://github.com/paritytech/polkadot-sdk/blob/a875f1803253e469288f1a905e73861fb600bb28/polkadot/runtime/parachains/src/configuration.rs#L1258.

Reply:
This is to protect against malformed runtime-injected values inside a config trait, not too much about user-provided values.

Reply:
The new solution does not really put a limit on the maximum denomination - as long as `10^denomination` can be represented in a U256 - but if the denomination is larger than the precision of the fixed, we will experience truncation of balance amounts during scaling, which I documented here.

Reply:
I also added an integrity test to check this, as suggested: 3a2306b615b0bab8fff0a07d5542e5e501c52287